contentprovider.go

```diff
@@ -339,6 +484,82 @@ func findSection(secs []DocumentSection, off, sz uint32) (int, bool) {
 	return 0, false
 }
 
```

Comment:
This part worries me the most: it's basically duplicating the core logic of the whole ranking.
However, I'm not sure how to factor this out and whether it is worth the effort.

Reply:
This is a very valid concern. It is also tough to review correctly, one almost wants to review the recipe you used to do copy paste here.

To be honest given this duplication I'm very close to recommending removal of the LineMatch field and only exposing the new type. This is a superior data model for what a match is and we don't have any guarantees around API (still not post v1 and likely never will be). WDYT?

Reply:
Alright, I made a [draft PR](https://github.com/sourcegraph/zoekt/pull/380) that takes the next step from this PR and removes `LineMatch` from everywhere but the API layer. I've based it off `master` rather than this PR so that the diff is actually useful and you can see the direct translation of line match logic to chunk match logic.